modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/directives/QueryDirectives.scala

```diff
@@ -47,13 +47,26 @@ trait QueryDirectives {
     }
 
   /**
-    * Extracts the pagination and filter query params from the request or defaults to preconfigured values.
+    * Extracts the ''term'' query param from the request.
+    */
+  def term: Directive1[Option[String]] =
+    parameter('term.?).flatMap(provide(_))
+
+  /**
+    * Extracts the ''deprecated'' query param from the request.
+    */
+  def deprecated: Directive1[Option[Boolean]] =
+    parameter('deprecated.as[Boolean].?).flatMap(provide(_))
```

Comment:
I would recommend avoiding this expressing functions like this: `provide(_)`.
Use `flatMap(opt => provide(opt))`.